sway/commands/font.c

```diff
@@ -16,12 +17,34 @@ struct cmd_results *cmd_font(int argc, char **argv) {
 	if (strncmp(font, "pango:", 6) == 0) {
 		config->pango_markup = true;
 		config->font = strdup(font + 6);
+		free(font);
 	} else {
 		config->pango_markup = false;
-		config->font = strdup(font);
+		config->font = font;
 	}
 
-	free(font);
+	// Parse the font early so we can reject it if it's not valid for pango.
+	// Also avoids re-parsing each time we render text.
+	PangoFontDescription *font_description = pango_font_description_from_string(config->font);
+
+	const char *family = pango_font_description_get_family(font_description);
+	if (family == NULL) {
+		pango_font_description_free(font_description);
+		return cmd_results_new(CMD_FAILURE, "Invalid font family.");
+	}
+
+	const gint size = pango_font_description_get_size(font_description);
+	if (size == 0) {
+		pango_font_description_free(font_description);
+		return cmd_results_new(CMD_FAILURE, "Invalid font size.");
+	}
+
+	if (config->font_description != NULL) {
+		pango_font_description_free(config->font_description);
+	}
+
+	config->font_description = font_description;
```

Comment:
This has a risk that the value will (for only an instant) be NULL, so if
another thread tries to read it, it'll fail. Could this be an issue? Should I
reassign _and then_ `free`?